### DISCUSSION TITLE
Is there a way to disable implicit conversion from float to int?

### DISCUSSION THREAD

**Discussion by advent94** [godotengine/godot-proposals#9319] [General Discussions]
```
const DURATION: float = 1.0
const COUNT: int = 1
const ARRAY: Array  = [DURATION, COUNT]

ARRAY[0] is float == false
```
In this example I am checking inside the script if ARRAY[0] is a float but since it's 1.0, it get's converted to int and my statement returns false. This ruins/requires additional corner case coverage for my script's implementation when I am checking if either I am dealing with float value or array of floats and all of them get converted to ints. Is there a way to stop this conversion or give a warning whenever it happens?

I checked Array[Variant] and it doesn't solve this issue.

**Comment by AThousandShips** [comment]
This doesn't seem like what's going on, can you do:
`print(typeof(ARRAY[0]), " ", TYPE_FLOAT)`? Accessing a member of an array doesn't provide static typing, so `is` might not be guaranteed to work, but there's no implicit casting to int in this case

**Reply by advent94** [reply]
Alright, it seems like I got some misunderstanding going on.

 It was different test that failed, one that didn't convert `Array = [ [1.0], [object]]`  to `Array = [ Array[float], Array[class] ]`.
 I understand now that this doesn't explicitly convert Array[Variant] to Array[type] so if I want to support Array[Variant] as argument, I need to check for types and if it has same typed object in implementation. Thank you for your input and sorry for bothering. By the way, I am really amazed and jealous seeing how much you contribute to open source community! If I hear any sexist remark related to programming, I will link people your profile 🤣

**Reply by AThousandShips** [reply]
Yes that's not happening automatically, that's not how the typed arrays work, you don't get typed arrays automatically, by design

See:
* https://github.com/godotengine/godot/issues/86963
* https://github.com/godotengine/godot-proposals/discussions/7364#discussioncomment-6913637
